trikot-kword/kword-remote-update/src/commonMain/kotlin/com/mirego/trikot/kword/remote/update/internal/InternalCacheWrapper.kt

```diff
@@ -0,0 +1,88 @@
+package com.mirego.trikot.kword.remote.update.internal
+
+import com.mirego.trikot.kword.DefaultI18N
+import com.mirego.trikot.kword.I18N
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.Json
+import okio.FileSystem
+import okio.IOException
+import okio.Path.Companion.toPath
+import okio.buffer
+
+class InternalCacheWrapper(
+    private val internalStoragePath: String?,
+    private val translationsVersion: String?,
+    private val fileSystem: FileSystem?
+) {
+    private val json = Json {
+        ignoreUnknownKeys = true
+        isLenient = true
+        allowSpecialFloatingPointValues = true
+    }
+
+    fun loadTranslationsFromCache(i18N: I18N, baseFileName: String, baseMap: MutableMap<String, String>, languageCodes: List<String>) {
+        fileSystem?.let { fileSystemVal ->
+            internalStoragePath?.let { internalStoragePathVal ->
+                translationsVersion?.let { translationsVersionVal ->
+                    if (internalStoragePathVal.isEmpty() || translationsVersionVal.isEmpty() || baseFileName.isEmpty()) {
+                        return
+                    }
+
+                    val baseDirectoryPath = "$internalStoragePathVal/translations/$translationsVersionVal".toPath()
+                    languageCodes.forEach { languageCode ->
+                        try {
+                            val fullSourcePath = baseDirectoryPath / buildFileName(baseFileName, languageCode)
+                            val source = fileSystemVal.source(fullSourcePath).buffer()
```

Comment:
⚠️  Seems like this is not closing the source stream at the end.
It'd be preferable to use this syntax which handles the closing
```
val result = fileSystemVal.read(path) {
    readUtf8()
}
```

Reply:
T'avais bien raison! Ya pu de leak avec cette syntax